### DISCUSSION TITLE
Avoiding stringified types

### DISCUSSION THREAD

**Discussion by bestouff** [tangramxyz/tangram#10] [Q&A] [answered]
Tangram looks fantastic for the ML beginners, thanks a lot !

Is there a way when using a pure Rust chain to use strong input types (e.g. `enum`, `f64` and `usize`) instead of only strings and floats ?

**Comment by nitsky** [comment, marked as answer]
Hi @bestouff thank you for opening the first discussion!

You absolutely can provide strong input types with the tangram library for rust. Check out [languages/rust/examples/advanced/main.rs](https://github.com/tangramxyz/tangram/blob/main/languages/rust/examples/advanced/main.rs) for an example. I am working on adding strong output types now as well, this will be available in the next release. In the future, we will also consider adding some code generation to automatically generate the type definitions from the model file. Let me know if there is anything else you are looking for!
